### PR TITLE
PHP-1510: Route unacknowledged writes through legacy op codes

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHP-1510

### CODE REVIEW

File: collection.c [hunk 1]
```diff
@@ -1622,7 +1631,11 @@ static void php_mongocollection_remove(zval *this_ptr, mongo_collection *c, zval
 		RETURN_FALSE;
 	}
 
-	if (php_mongo_api_connection_supports_feature(connection, PHP_MONGO_API_WRITE_API)) {
+	is_acknowledged = is_gle_op(this_ptr, z_write_options, &link->servers->options TSRMLS_CC);
+	supports_command = php_mongo_api_connection_supports_feature(connection, PHP_MONGO_API_WRITE_API);
+	supports_opcode = php_mongo_api_connection_supports_feature(connection, PHP_MONGO_API_RELEASE_2_4_AND_BEFORE);
```
Comment (derickr) [inline collection.c]: As you're doing this 3 times - would it make sense to split it up in a method call with three "out" variables? Something like:

```
static void figure_out_support(this_ptr, z_write_options, link->server->options, connection, &is_acknowledged, &supports_command, &supports_opcode);
```

File: collection.c [hunk 2]
```diff
@@ -1134,7 +1135,11 @@ static void php_mongo_collection_insert(zval *z_collection, zval *document, zval
 		RETURN_FALSE;
 	}
 
-	if (php_mongo_api_connection_supports_feature(connection, PHP_MONGO_API_WRITE_API)) {
+	is_acknowledged = is_gle_op(z_collection, z_write_options, &link->servers->options TSRMLS_CC);
```
Comment (derickr) [inline collection.c]: I like that you picked ``is_acknowledged`` and not ``is_gle_op``. I was going make sure after reading the truth table in the ticket, but you already did it :-)

**Comment by derickr** [comment]
LGTM, after we:

 - [ ] Spoke about the possible functification.

File: collection.c (hunk 2 above)
Comment (jmikola) [inline collection.c]: I originally used `is_gle_op`, but it was a symbol conflict with the static function name :laughing:

File: collection.c (hunk 1 above)
Comment (jmikola) [inline collection.c]: I'm not too inclined to create a new function, as we already have a bunch of static helpers in this file. I'm open to a macro for the conditional expression or adding a block comment above each of the three conditionals to explain the reasoning.

File: collection.c (hunk 2 above)
Comment (derickr) [inline collection.c]: hehe :-)

**Pull request merged**